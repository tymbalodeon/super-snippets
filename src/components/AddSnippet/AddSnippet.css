.AddSnippet * > button {
  margin-top: 20px;
  margin-left: 5px;
  margin-right: 5px;
  padding: 5px;
}

.AddSnippet > h2 {
  color: #5ccfe6;
  text-align: center;
}

input,
textarea {
  width: 100%;
  font-size: 16px;
}

textarea {
  height: 200px;
}

.buttons {
  text-align: center;
}

.AddSnippet * > button[type='submit'] {
  background-color: #ffa759;
  border-color: #ffa759;
}
