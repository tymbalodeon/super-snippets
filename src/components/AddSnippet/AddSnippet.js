import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import SnippetApiService from '../../services/snippet-api-service';
import './AddSnippet.css';

export default class AddSnippet extends Component {
  handleSubmit = e => {
    e.preventDefault();
    const { title, info, content, project_id } = e.target;
    const newSnippet = {
      title: title.value,
      info: info.value,
      content: content.value,
      project_id: project_id.value
    };
    const { addSnippet } = this.props;
    let id;
    SnippetApiService.postSnippet(newSnippet)
      .then(res => (id = res.id))
      .then(addSnippet)
      .then(() => {
        title.value = '';
        info.value = '';
        content.value = '';
        project_id.value = null;
      })
      .then(() => this.props.history.push(`/snippets/${id}`))
      .catch();
  };

  componentDidMount() {
    SnippetApiService.getProjects()
      .then(res => this.props.setProjects(res))
      .catch(error => this.setState({ error }));
  }

  render() {
    const { projects, error } = this.props;
    return (
      <section className="AddSnippet">
        <h2>Create a snippet</h2>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <form onSubmit={this.handleSubmit}>
          <div className="field">
            <label htmlFor="snippet-name-input">Name: </label>
            <input type="text" id="snippet-name-input" name="title" />
          </div>
          <div className="field">
            <label htmlFor="snippet-info-input">Description: </label>
            <input type="text" id="snippet-info-input" name="info" />
          </div>
          <div className="field">
            <label htmlFor="snippet-content-input">Content: </label>
            <textarea id="snippet-content-input" name="content" />
          </div>
          <div className="field">
            <label htmlFor="snippet-project-select">Project: </label>
            <select id="snippet-project-select" name="project_id">
              <option value={null}>--Choose A Project--</option>
              {projects.map(project => (
                <option key={project.id} value={project.id}>
                  {project.title}
                </option>
              ))}
            </select>
          </div>
          <div className="buttons">
            <Link to="#" onClick={this.props.history.goBack}>
              <button className="back">BACK</button>
            </Link>
            <button type="submit">ADD</button>
          </div>
        </form>
      </section>
    );
  }
}
