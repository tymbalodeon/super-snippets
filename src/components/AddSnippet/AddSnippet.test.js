import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import renderer from 'react-test-renderer';
import AddSnippet from './AddSnippet';
import testSeed from '../../test-seed';

describe.skip('<AddSnippet />', () => {
  const history = createMemoryHistory('/');
  const projects = testSeed.makeProjects();

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <AddSnippet history={history} />
      </BrowserRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    console.log(projects);
    const tree = renderer
      .create(
        <BrowserRouter>
          <AddSnippet history={history} projects={projects} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
