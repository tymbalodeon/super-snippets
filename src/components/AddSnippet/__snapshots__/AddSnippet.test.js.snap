// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddSnippet /> renders the UI as expected 1`] = `
<section
  className="AddSnippet"
>
  <h2>
    Create a snippet
  </h2>
  <div
    role="alert"
  />
  <form
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label
        htmlFor="snippet-name-input"
      >
        Name: 
      </label>
      <input
        id="snippet-name-input"
        name="title"
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="snippet-info-input"
      >
        Description: 
      </label>
      <input
        id="snippet-info-input"
        name="info"
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="snippet-content-input"
      >
        Content: 
      </label>
      <textarea
        id="snippet-content-input"
        name="content"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="snippet-project-select"
      >
        Project: 
      </label>
      <select
        id="snippet-project-select"
        name="project_id"
      >
        <option
          value={null}
        >
          --Choose A Project--
        </option>
        <option
          value={1}
        >
          Project with Live Trumpet
        </option>
        <option
          value={2}
        >
          Example
        </option>
      </select>
    </div>
    <div
      className="buttons"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="back"
        >
          BACK
        </button>
      </a>
      <button
        type="submit"
      >
        ADD
      </button>
    </div>
  </form>
</section>
`;
